wordpress-mobile/WordPress-iOS
# Post Settings: Remove ability to set post location

Fixes #16234

## Description

This PR removes the Location feature under Post Settings.

Geotagging posts (setting their location) [is no longer supported on the web](https://wordpress.com/support/replacing-the-older-wordpress-com-editor-with-the-wordpress-block-editor/) and was [removed from WPAndroid](https://github.com/wordpress-mobile/WordPress-Android/pull/13378) about six months ago.

Before | After
-- | --
![Simulator Screen Shot - iPhone 12 Pro - 2021-08-18 at 17 46 51](https://user-images.githubusercontent.com/6711616/129940101-378fc3b5-6832-4d15-a573-bc8c3ce5a722.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-08-18 at 17 54 31](https://user-images.githubusercontent.com/6711616/129940103-0d0865e1-9238-4943-a24a-11d1c3710aac.png)

## To test

1. Install an older version of the app, without the changes from this PR
2. Check out this branch, then run and install the app
3. ✅ Should run as normal -- this means that the lightweight Core Data migration is working as expected
4. Open **Post Settings** on a post or page
5. ✅  Notice that the option to **set the location** is no longer there


## Regression Notes
1. Potential unintended areas of impact
Post Settings

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I built and ran the post settings screen

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:
